modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -1518,9 +1520,16 @@ else if (res.resultType() == ResultType.LOCKED) {
 
                 CacheLazyEntry e = new CacheLazyEntry(cctx, key, old, keepBinary);
 
-                Object interceptorVal = cctx.config().getInterceptor().onBeforePut(
-                    new CacheLazyEntry(cctx, key, old, keepBinary),
-                    val0);
+                Object interceptorVal = val0;
+
+                if(cctx.dr().cacheInterceptorDisabled() && explicitVer != null && explicitVer.dataCenterId() != cctx.dr().dataCenterId())
```

Comment:
This could be simpified to
`if (!(interceptDisabled = cctx.dr().cacheInterceptorDisabled() && explicitVer != null &&
                    explicitVer.dataCenterId() != cctx.dr().dataCenterId()))
                    interceptorVal = cctx.config().getInterceptor().onBeforePut(
                        new CacheLazyEntry(cctx, key, old, keepBinary),
                        val0);`

Also you should avoid too long lines and keep right formatting.